lib/net/http/request.gb

```diff
@@ -1,7 +1,22 @@
 module Net
   class HTTP
     class Request
-      attr_accessor :body, :method, :path, :url, :host
+      attr_accessor :method, :protocol, :body, :content_length, :transfer_encoding, :host, :path, :url
+      attr_reader   :headers
+
+      def set_header(key, value)
+        if @headers == nil
+          @headers = {}
+        end
+        @headers[key] = value
+      end
+
+      def get_header(key)
+        return @headers[key]
+      end
+
+      def delete_header(key)
+      end
```

Comment:
The `delete_header` method should be use the `Hash#delete` method which isn't implemented yet but stated in #247